src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriterCache.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines a thread-local cache for JsonSerializer to store reusable Utf8JsonWriter/IBufferWriter instances.
+    /// </summary>
+    internal static class Utf8JsonWriterCache
+    {
+        [ThreadStatic]
+        private static ThreadLocalState? s_threadLocalState;
```

Comment:
How much memory might this be consuming?  This can mean that any thread which does any serialization now increases in size by this amount, which is a form of a leak.  We thus try to be careful about how large this state can be.  Is there a limit to it?